File: quarkus3/src/main/java/org/acme/repository/FruitRepository.java
package org.acme.repository;

import java.util.Optional;

import jakarta.enterprise.context.ApplicationScoped;

import org.acme.domain.Fruit;

import io.quarkus.hibernate.orm.panache.PanacheRepository;

@ApplicationScoped
public class FruitRepository implements PanacheRepository<Fruit> {
	public Optional<Fruit> findByName(String name) {
		return find("name", name).firstResultOptional();
	}
}

File: dotnet10/Dotnet10/Repository/FruitRepository.cs
using Dotnet10.Data;
using Dotnet10.Domain;
using Microsoft.EntityFrameworkCore;

namespace Dotnet10.Repository;

/// <summary>
/// EF Core implementation of IFruitRepository.
/// Equivalent to FruitRepository extends PanacheRepository&lt;Fruit&gt; in the Quarkus project.
/// Eager loading of StorePrices and their Stores mirrors Hibernate FetchType.EAGER / FetchMode.SELECT.
/// </summary>
public class FruitRepository(AppDbContext context) : IFruitRepository
{
    public async Task<List<Fruit>> ListAllAsync() =>
        await context.Fruits
            .Include(f => f.StorePrices)
            .ThenInclude(sp => sp.Store)
            .OrderBy(f => f.Id)
            .ToListAsync();

    public async Task<Fruit?> FindByNameAsync(string name) =>
        await context.Fruits
            .Include(f => f.StorePrices)
            .ThenInclude(sp => sp.Store)
            .FirstOrDefaultAsync(f => f.Name == name);

    public async Task PersistAsync(Fruit fruit)
    {
        context.Fruits.Add(fruit);
        await context.SaveChangesAsync();
    }
}
